cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1801,6 +1783,81 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 	return nil
 }
 
+func installNewSystemGenerateMounts(mst *initramfsMountsState) error {
+	start := time.Now()
+	logger.Noticef("Installing new system...\n")
+
+	systemLabel := mst.recoverySystem
+	// 1.0 mount ubuntu-seed
+	if err := mountNonDataPartitionMatchingKernelDisk(boot.InitramfsUbuntuSeedDir, "ubuntu-seed"); err != nil {
+		return err
+	}
+
+	activeMounts = append(activeMounts, mountInfo{
+		mountPoint: boot.InitramfsUbuntuSeedDir,
+		unmountFnc: func() error {
+			sd := systemd.New(systemd.SystemMode, progress.Null)
+			return sd.Umount(boot.InitramfsUbuntuSeedDir)
+		},
+		essential: false})
+
+	// 1.1 load seed
+	deviceSeed, err := loadDeviceSeed(systemLabel)
+	if err != nil {
+		return err
+	}
+	// 1.2 load seed assertions
+	assertDb, err := loadSeedAssertions(deviceSeed)
+	if err != nil {
+		return err
+	}
+
+	// 1.3 load sead essential meta
+	if err = loadSeedEssentialMeta(deviceSeed, systemLabel); err != nil {
+		return err
+	}
+
+	// 1.4 mount gadget and kernel snap
+	if err = mountEssentials(deviceSeed, []snap.Type{snap.TypeGadget, snap.TypeKernel}); err != nil {
+		return err
+	}
+
+	// 2.0 run install
+	if err = doRunInstall(deviceSeed, systemLabel); err != nil {
+		return err
+	}
+
+	// 2.1 setup presseded data if needed
+	if err = maybeApplyPreseededData(deviceSeed, assertDb, systemLabel); err != nil {
+		return err
+	}
+
+	// 2.2 ensure next boot is to run mode
+	if err := boot.EnsureNextBootToRunMode(systemLabel); err != nil {
```

Comment:
there is a bit more than this happening in doRestartSystemToRunMode that we might want/need to take care of